src/test/java/io/odh/test/e2e/standard/UninstallST.java

```diff
@@ -63,56 +66,75 @@ public class UninstallST extends StandardAbstract {
+        if (ResourceManager.getKubeCmdClient().namespace(OdhConstants.OLM_OPERATOR_NAMESPACE).list(
+                "configmap").contains(DELETE_CONFIG_MAP_NAME)) {
+            Assertions.fail(
+                    String.format("The ConfigMap '%s' is present on the cluster before the uninstall test started!",
+                            DELETE_CONFIG_MAP_NAME));
         }
 
+        ConfigMap cm = new ConfigMapBuilder()
+                .withNewMetadata()
+                .withName(DELETE_CONFIG_MAP_NAME)
+                .withNamespace(OdhConstants.OLM_OPERATOR_NAMESPACE)
+                .withLabels(Map.ofEntries(Map.entry(DELETE_ANNOTATION, "true")))
+                .endMetadata()
+                .build();
+        ResourceManager.getInstance().createResourceWithWait(cm);
+
         // Now the product should start to uninstall, let's wait a bit and check the result.
         TestUtils.waitFor(String.format("the '%s' namespace to be removed as operator is being uninstalled",
-                        OdhConstants.CONTROLLERS_NAMESPACE), 2000, 20000,
+                        OdhConstants.CONTROLLERS_NAMESPACE), 2000, 120_000,
                 () -> !ResourceManager.getKubeClient().namespaceExists(OdhConstants.CONTROLLERS_NAMESPACE));
 
+        // Operator itself should delete the CSV, Subscription and InstallPlan
+        Assertions.assertTrue(ResourceManager.getKubeCmdClient().namespace(OdhConstants.OLM_OPERATOR_NAMESPACE).list(
+                "subscriptions").isEmpty(), "The operator Subscription is still present!");
+        Assertions.assertTrue(ResourceManager.getKubeCmdClient().namespace(OdhConstants.OLM_OPERATOR_NAMESPACE).list(
+                "installplan").isEmpty(), "The operator InstallPlan is still present!");
+        Assertions.assertFalse(ResourceManager.getKubeCmdClient().namespace(OdhConstants.OLM_OPERATOR_NAMESPACE).list(
+                "csv").stream().anyMatch(s -> s.toString().contains(OdhConstants.OLM_OPERATOR_NAME)),
+                "The operator CSV is still present!");
+
+        // TODO Check that the config map is deleted also
+        // Looks like operator don't touch this.
+        // This is deleted together with the operator namespace deletion for RHOAI, but what about ODH?
+//        Assertions.assertFalse(ResourceManager.getKubeCmdClient().namespace(OdhConstants.OLM_OPERATOR_NAMESPACE).list("configmap").stream().anyMatch(s -> s.toString().equals(DELETE_CONFIG_MAP_NAME)),
+//                "The operator deletion ConfigMap is still present!");
+
         // Let's remove the operator namespace now
-        ResourceManager.getKubeCmdClient().deleteNamespace(OdhConstants.OLM_OPERATOR_NAMESPACE);
-        NamespaceUtils.waitForNamespaceDeletion(OdhConstants.OLM_OPERATOR_NAMESPACE);
+        if (Environment.PRODUCT.equals(Environment.PRODUCT_ODH)) {
+            LOGGER.info(String.format("Tested product is ODH - skipping removal of the '%s' namespace.", OdhConstants.OLM_OPERATOR_NAMESPACE));
```

Comment:
This will never run, due to the @DisabledIf, no?

Reply:
Yes, not now until we resolve this for ODH.